mlflow/pipelines/steps/split.py

```diff
@@ -109,6 +109,45 @@ def _create_hash_buckets(input_df):
     return hash_buckets
 
 
+def _validate_user_code_output(post_split, train_df, validation_df, test_df):
+    try:
+        (
+            post_filter_train_df,
+            post_filter_validation_df,
+            post_filter_test_df,
+        ) = post_split(train_df, validation_df, test_df)
+    except Exception:
+        raise MlflowException(
+            message="Error in cleaning up the data frame post split step."
+            " Expected output is a list with (train_df, validation_df, test_df)"
+        ) from None
+
+    import pandas as pd
+
+    for (post_split_df, pre_split_df, split_type) in [
+        [post_filter_train_df, train_df, "train"],
+        [post_filter_validation_df, validation_df, "validation"],
+        [post_filter_test_df, test_df, "test"],
+    ]:
+        if not isinstance(post_split_df, pd.DataFrame):
+            raise MlflowException(
+                message="The split data is not a DataFrame, please return the correct data."
+            ) from None
+        if len(pre_split_df.columns) != len(post_split_df.columns):
```

Comment:
Instead of matching the size of columns, can we match on the column name instead?
`if list(pre_split_df.columns) != list(post_split_df.columns):`